src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java

```diff
@@ -331,21 +332,34 @@ void setShuntCompensators(List<ShuntCompensator> shuntCompensators, LfNetworkPar
         if (!parameters.isShuntVoltageControl() && !shuntCompensators.isEmpty()) {
             shunt = new LfShuntImpl(shuntCompensators, network, this, false, parameters, topoConfig);
         } else {
-            List<ShuntCompensator> controllerShuntCompensators = shuntCompensators.stream()
-                    .filter(ShuntCompensator::isVoltageRegulatorOn)
-                    .toList();
+            Map<Boolean, List<ShuntCompensator>> shuntsByVoltageControlCapability = shuntCompensators.stream()
+                    .collect(Collectors.partitioningBy(shunt -> checkVoltageRegulation(shunt, parameters)));
+
+            List<ShuntCompensator> controllerShuntCompensators = shuntsByVoltageControlCapability.get(true);
             if (!controllerShuntCompensators.isEmpty()) {
                 controllerShunt = new LfShuntImpl(controllerShuntCompensators, network, this, true, parameters, topoConfig);
             }
-            List<ShuntCompensator> fixedShuntCompensators = shuntCompensators.stream()
-                    .filter(sc -> !sc.isVoltageRegulatorOn())
-                    .toList();
+            List<ShuntCompensator> fixedShuntCompensators = shuntsByVoltageControlCapability.get(false);
             if (!fixedShuntCompensators.isEmpty()) {
                 shunt = new LfShuntImpl(fixedShuntCompensators, network, this, false, parameters, topoConfig);
             }
         }
     }
 
+    static boolean checkVoltageRegulation(ShuntCompensator shuntCompensator, LfNetworkParameters parameters) {
+        double nominalV = shuntCompensator.getRegulatingTerminal().getVoltageLevel().getNominalV();
+        double targetV = shuntCompensator.getTargetV();
+        if (!shuntCompensator.isVoltageRegulatorOn()) {
+            return false;
+        }
+        if (!VoltageControl.checkTargetV(targetV / nominalV, nominalV, parameters)) {
+            shuntCompensator.setVoltageRegulatorOn(false);
```

Comment:
don't modify the iIDM Network, remove this line. the regulator is disabled only internally in OLF.